packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+
+## Espressive CRUD Table
+
+Given the nature of our actual tables and the operations we need to perform on the data entities, a CRUD Table starts to make sense. A kind of table that not only displays data in a nice tabular way, but also allow for a richier interaction with the data.
+
+It is true that CRUD operations are needed in less scenarios, so the CRUD feature must be easily enabled and must not interfere with the regular table logic.
+
+## API
+
+The API has been divided into
+
+### Events
+
+In order to support CRUD operations, we must define a mechanism to communicate the parent any changes to the data.
+
+- onCreate
+- onUpdate
+- onDelete
+- onCustomAction
+
+Please note that the names could have been something like `onItemCreate` for semantics, but let's remember that in the future we plan to support selection and custom actions on multiple items. Our proposal is to keep these as abstract as possible.
+
+| Event          | Signature                                                  | Description                                                     |
+| -------------- | ---------------------------------------------------------- | --------------------------------------------------------------- |
+| onCreate       | `(Object: newItem) => {}`                                  | Fired when the User saves a new item                            |
+| onUpdate       | `(Array[Object]: updatedItems) => {}`                      | Fired when one or more rows have been updated by the User       |
+| onDelete       | `(Array[String]: selectedItems) => {}`                     | Fired when one or more rows have been deleted by the User       |
+| onCustomAction | `(String: actionType, Array[String]: selectedItems) => {}` | Fired when a custom action has been applied to one or more rows |
+
+### Config
+
+There will be use cases where features like multi-selection, CRUD, custom actions, etc. won't be required at all, they must be disabled by thefault but easily activated. It makes sense to have boolean props that controll such features.
+
+Another important aspect of configuration is the definition of columns, hence a complete section is devoted to it.
+
+| Name    | Type          | default | Description                                        |
+| ------- | ------------- | ------- | -------------------------------------------------- |
+| Bulk    | Boolean       | `FALSE` | If true, activates the bulk mode (multi-select)    |
+| Columns | Array[Object] | `[]`    | Holds the definitions for each column in the table |
```

Comment:
When I got the the columns section below, I realized that if we are going to use consistent patterns in Cascara for how some of our props work... that might mean we want to name columns differently here. That way if I look at this prop as a developer, if I have seen and used it before on another Cascara component, I already know how it works here.

Maybe a better name for this is `dataConfig`. This would keep it sorted alphabetically with the `data` prop that it controls (and I love that from an OCD perspective), but that also would make it so we could use a `dataConfig` prop on any other Cascara components that use the same `data` > `dataConfig` > `doMagic` pattern. The objects expected in `dataConfig` could be consistent everywhere. If we are working with a List, Form, Table... if we see a `dataConfig` prop, we know that it is going to control how the data looks in the component we are applying it to, and it should have mostly the same object shape of the objects in the array.

I guess in theory... if we have a Table with a `data` and `dataConfig` defined, we might be able to swap it over to a List component instead, and except for a few minor tweaks to parameters on the `dataConfig` controlling where the data shows on the list items, it would immediately work.